### DISCUSSION TITLE
when i use enableGarbageCollection: true , milvus return err as the follow.

### DISCUSSION THREAD

**Discussion by bkwcl** [milvus-io/milvus#17189] [Q&A and General discussion]
milvus-standalone | [2022/05/24 08:20:59.068 +00:00] [DEBUG] [client.go:83] ["DataCoordClient, not existed in msess "] [key=datacoord] ["len of msess"=0]
milvus-standalone | [2022/05/24 08:20:59.068 +00:00] [ERROR] [client.go:131] ["failed to get client address"] [error="find no available datacoord, check datacoord state"] [stack="github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase).connect\n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:131\ngithub.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase).GetGrpcClient\n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:103\ngithub.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase).callOnce\n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:193\ngithub.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase).ReCall\n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:251\ngithub.com/milvus-io/milvus/internal/distributed/datacoord/client.(*Client).GetComponentStates\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/datacoord/client/client.go:111\ngithub.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates.func1\n\t/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:59\ngithub.com/milvus-io/milvus/internal/util/retry.Do\n\t/go/src/github.com/milvus-io/milvus/internal/util/retry/retry.go:34\ngithub.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates\n\t/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:83\ngithub.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentInitOrHealthy\n\t/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:88\ngithub.com/milvus-io/milvus/internal/distributed/datanode.(*Server).init\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/datanode/service.go:288\ngithub.com/milvus-io/milvus/internal/distributed/datanode.(*Server).Run\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/datanode/service.go:165\ngithub.com/milvus-io/milvus/cmd/components.(*DataNode).Run\n\t/go/src/github.com/milvus-io/milvus/cmd/components/data_node.go:50\ngithub.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).runDataNode.func1\n\t/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:275"]
milvus-standalone | [2022/05/24 08:20:59.068 +00:00] [ERROR] [client.go:254] ["datacoord ClientBase ReCall grpc second call get error "] [error="err: find no available datacoord, check datacoord state\n, /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace\n/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:253 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase).ReCall\n/go/src/github.com/milvus-io/milvus/internal/distributed/datacoord/client/client.go:111 github.com/milvus-io/milvus/internal/distributed/datacoord/client.(*Client).GetComponentStates\n/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:59 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates.func1\n/go/src/github.com/milvus-io/milvus/internal/util/retry/retry.go:34 github.com/milvus-io/milvus/internal/util/retry.Do\n/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:83 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates\n/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:88 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentInitOrHealthy\n/go/src/github.com/milvus-io/milvus/internal/distributed/datanode/service.go:288 github.com/milvus-io/milvus/internal/distributed/datanode.(*Server).init\n/go/src/github.com/milvus-io/milvus/internal/distributed/datanode/service.go:165 github.com/milvus-io/milvus/internal/distributed/datanode.(*Server).Run\n/go/src/github.com/milvus-io/milvus/cmd/components/data_node.go:50 github.com/milvus-io/milvus/cmd/components.(*DataNode).Run\n"] [stack="github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase).ReCall\n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:254\ngithub.com/milvus-io/milvus/internal/distributed/datacoord/client.(*Client).GetComponentStates\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/datacoord/client/client.go:111\ngithub.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates.func1\n\t/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:59\ngithub.com/milvus-io/milvus/internal/util/retry.Do\n\t/go/src/github.com/milvus-io/milvus/internal/util/retry/retry.go:34\ngithub.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates\n\t/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:83\ngithub.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentInitOrHealthy\n\t/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:88\ngithub.com/milvus-io/milvus/internal/distributed/datanode.(*Server).init\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/datanode/service.go:288\ngithub.com/milvus-io/milvus/internal/distributed/datanode.(*Server).Run\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/datanode/service.go:165\ngithub.com/milvus-io/milvus/cmd/components.(*DataNode).Run\n\t/go/src/github.com/milvus-io/milvus/cmd/components/data_node.go:50\ngithub.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).runDataNode.func1\n\t/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:275"]

[the remark this answers is not in the document]
**Reply by wayblink** [reply]
Hi, @bkwcl ！Thanks for the feedback. According to the log, I can't find any info about garbage collection. It was more like the data_coord component was not registered successfully to etcd. Is this issue reproducible？Does it fail only when we set enableGarbageCollection=false? If true, could you upload the complete log, so that we can investigate it.

[the remark this answers is not in the document]
**Reply by ndtreviv** [reply]
I have the same issue. Here is my log with the IP redacted
[milvus-log.txt](https://github.com/milvus-io/milvus/files/9395085/milvus-log.txt)
I hope it helps!

[the remark this answers is not in the document]
**Reply by ndtreviv** [reply]
PS: I changed no settings at all. The system crashed and milvus wouldn't come back up again.

[the remark this answers is not in the document]
**Reply by wayblink** [reply]
> PS: I changed no settings at all. The system crashed and milvus wouldn't come back up again.

Will take a look later. What milvus version you are using?

[the remark this answers is not in the document]
**Reply by ndtreviv** [reply]
milvus v2.0.2, minio/minio:RELEASE.2020-12-03T00-03-10Z, quay.io/coreos/etcd:v3.5.0

[the remark this answers is not in the document]
**Reply by wayblink** [reply]
Please use the latest 2.1.X version if you can, a lot of changes have been made between 2.0.2 and 2.1. Perhaps it can be solved by upgrade

[the remark this answers is not in the document]
**Reply by zhongqingyang** [reply]
I encountered the same problem on 2.1.4

[the remark this answers is not in the document]
**Reply by wayblink** [reply]
> I encountered the same problem on 2.1.4

Hi, you can open a new issue and upload logs so that we can investigate it.

[the remark this answers is not in the document]
**Reply by zhongqingyang** [reply]
I deleted an image by mistake and solved it after reloading

**Comment by ndtreviv** [comment]
I upgraded to milvus 2.1.1 and the standalone milvus is still the weakest link in this setup. It brings everything down regularly to the point where the machine often needs a hard reset. 

I've attached another set of logs for review.
[milvus_1.log](https://github.com/milvus-io/milvus/files/9880772/milvus_1.log)
[milvus_2.log](https://github.com/milvus-io/milvus/files/9880773/milvus_2.log)
